comm/src/main/scala/coop/rchain/comm/network.scala

```diff
@@ -153,21 +152,23 @@ class UnicastNetwork(peer: PeerNode) {
   private def handleResponse[F[_]: Monad: Capture: Log: Time: Metrics](
       sock: SocketAddress,
       sender: PeerNode,
-      msg: ProtocolResponse): F[Option[(SocketAddress, ProtocolMessage)]] = {
-    val handleWithHeader: Option[F[Option[(SocketAddress, ProtocolMessage)]]] =
-      msg.returnHeader.map { ret =>
-        for {
-          result <- Capture[F].capture(pending.get(PendingKey(sender.key, ret.timestamp, ret.seq)))
-          ret <- result match {
-                  case Some(promise) => Capture[F].capture(promise.success(Right(msg))).as(None)
-                  case None          => Some((sock, msg)).pure[F]
-                }
+      msg: ProtocolResponse): F[Option[ProtocolMessage]] = {
+    val handleWithHeader: Option[F[Option[ProtocolMessage]]] = for {
+      ret  <- msg.returnHeader
```

Comment:
Could you rename `ret` to `returnHeader` so it's easier to distinguish (at first glance) from `ret` on line 162?